### PR TITLE
Debugged load.py

### PR DESCRIPTION
**Pull request by hhm970**
- Debugged load.py
- Wrote unit tests for functions in load.py

### CODE REVIEW

File: pipeline/load/conftest.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import pytest
+
+
+@pytest.fixture
```
Comment (shindym) [inline pipeline/load/conftest.py:4]: Good use of conftest.py

File: pipeline/load/load.py [hunk 2]
```diff
@@ -81,48 +81,50 @@ def input_game_tags_into_db(game_data: list[list], conn: connection) -> None:
...
+                    tag_formatted = tag.title()
+                    cur.execute("""SELECT tag_id FROM tag
+                                WHERE SIMILARITY(%s, tag_name) > 0.8""",
+                                (tag_formatted,))
+                    tag_id_match = cur.fetchone()
 
-                if tag_id_match is None:
-                    cur.execute("""INSERT INTO tag (tag_name)
-                                VALUES %s""", (tag_formatted,))
+                    if tag_id_match is None:
```
Comment (shindym) [inline pipeline/load/load.py:95]: good use of checking if the tag already exists.

File: pipeline/load/load.py [hunk 3]
```diff
@@ -81,48 +81,50 @@ def input_game_tags_into_db(game_data: list[list], conn: connection) -> None:
...
+                    tag_id = tag_id_match['tag_id']
+
+                    cur.execute("""INSERT INTO game_tag_matching (game_id, tag_id)
+                                VALUES (%s, %s)""", (game_id, tag_id))
         cur.close()
     conn.commit()
 
 
-def handler(event=None, context=None) -> None:
+def handler(event: list[list[list]] = None, context=None) -> None:
```
Comment (shindym) [inline pipeline/load/load.py:112]: Good use of typehints.

File: pipeline/load/test_load.py [hunk 4]
```diff
@@ -1,3 +1,28 @@
...
+from datetime import datetime
 
-from load import get_db_connection, format_release_date_dt
+from unittest.mock import patch, MagicMock
+
+from load import (format_release_date_dt, input_game_into_db,
+                  input_game_plat_into_db, input_game_tags_into_db)
+
+
+def test_format_release_date_dt(test_game_data):
```
Comment (shindym) [inline pipeline/load/test_load.py:10]: Add docstrings to the tests.

Comment (hhm970) [reply pipeline/load/test_load.py:10]: Thanks, I've added them in!

**Review by shindym — APPROVED** [review]
Thanks for the changes!

**Pull request merged**